text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,2195 @@
+> of the metrics to be sent is known in advance.
+
+### Logs
+
+#### Batch Size Analysis
+
+The dataset used for this benchmark contains 200K logs with 43 different structures (i.e. combination of different
+fields).
+
+The following multi-line plot show the compressed batch size in bytes for different protocols and combination of
+parameters:
+
+* protocol (OTLP or OTLP Arrow)
+* batch size (5000, 10000, 25000, 50000, 100000)
+* compression algorithms (Zlib, Lz4, Zstd)
+* Sorted vs unsorted columns (only for OTLP Arrow)
+
+![Metrics bytes](img/0156_logs_bytes.png)
+
+The best results are obtained with OTLP Arrow + text column sorting + Zstd. This corroborates the results in the
+columnar
+database domain. The compression algorithm is much better on columns that are sorted or partially sorted.
+
+In the case of logs, the Zstd compression algorithm is the best for both OTLP and OTLP Arrow.
+
+> The compression ratio for OTLP Arrow is about twice that of OTLP.
+
+#### Serialization/Compression Time Analysis
+
+This section compares the time spent creating, serializing, compressing, decompressing and deserializing batches of
+metrics with OTLP and OTLP Arrow while looking to optimize message size. In both cases, the data producer delivers the
+logs in a row-oriented approach. The Arrow OTLPs therefore integrate the row-to-column conversion time.
+
+The following stacked bar plot compares side by side the distribution of time spent for each step and for each version
+of the protocol.
+
+![Breakdown of time per step for each batch size and each protocol](img/0156_logs_step_times.png)
+
+When cumulating the times of all steps, OTLP Arrow (with sorted columns) is ~0.90 slower than standard OTLP. This is
+mainly due to the ordering of the text columns. Without column sorting, OTLP Arrow is 1.3 times
+faster than OTLP. For bandwidth optimization it is therefore recommended, at the cost of a slight processing slowdown,
+to activate column sorting (see
+appendix [Parameter tuning and design optimization](#appendix-c---parameter-tuning-and-design-optimization)
+for more details on the parameters to select the columns to sort).
+
+The batch creation+sorting phase for OTLP Arrow represents almost all the time spent. Serialization and deserialization
+times are
+almost zero due to the use of Flatbuffer by Apache Arrow (ser/deser without parsing/unpacking). Compression and
+decompression
+times are low due to the fact that the size of the Arrow OTLP message before compression is more than 7 times
+smaller than an OTLP message with the same content.
+
+> It should be possible to significantly optimize the creation of OTLP Arrow batches for contexts where the structure
+> of the logs to be sent is known in advance.
+
+### Traces
+
+No benchmark has been conducted for traces due to lack of a dataset containing traces with links and events.
+
+Please feel free to share a dataset with such characteristics to complete this document.
```

Comment:
On the topic of data sharing, I believe it is time for OTel to maintain an official, open-source tool suite to enable easy exchange of obfuscated telemetry data: see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13626. For the record, Lightstep produced a single dataset and shared with @lquerel -- but it has zero Span Links and this sort of tooling will help us solicit input from the wider community.)

Reply:
Added a link to this issue in the OTEP.